instrumentation/active_record/lib/opentelemetry/instrumentation/active_record/patches/querying.rb

```diff
@@ -18,9 +18,17 @@ class << base
 
           # Contains ActiveRecord::Querying to be patched
           module ClassMethods
-            def find_by_sql(...)
-              tracer.in_span("#{self}.find_by_sql") do
-                super
+            if ::ActiveRecord.version >= Gem::Version.new('7.0.0')
+              def _query_by_sql(...)
+                tracer.in_span("#{self}.find_by_sql") do
```

Comment:
I get what you're saying about the naming! If we change the name of the span, we should consider marking this update as a breaking change since all 7.0 spans would now have a new name.

I like your suggestion of `query` for 7.0+ spans since they don't roll up to `find_by_sql`. 

Unfortunately, instrumentation often has to resort to private methods to get the data users want, so `_query_by_sql` probably wouldn't be too much of a shock and perhaps easier for a user to connect the dots. 

I'd love to hear other opinions!

Reply:
I feel like `query` (or similar) would be the most stable for the future. Since we're hooking onto internal AR methods, it's possible we might need to change methods, needing a breaking change every time.

The downside is that all other spans being generated have the name of the method we're hooking into, so it would be a bit inconsistent. We could make it configurable?

As a user of this instrumentation I'd lean towards something more generic and stable. I usually have the `mysql2` or `trilogy` spans with details about the exact query that ran, and as a developer I'd need to open the rails code to know what `_query_by_sql` even is. Even the `find_by_sql` was a bit confusing, I didn't know all reads (used to) go through that method.

Reply:
Nice, I'm leaning toward `query` now too!

I hesitate to make it configurable. We have some configuration around span naming for ActiveSupport::Notifications in some of our Rails-related instrumentation gems, but in this scenario, since the span is not related to ActiveSupport::Notifications, I think having a single, consistent name would be best.

Reply:
I pushed a change to use `query` instead of `find_by_sql`. Spans are now called `User query`, `Account query`, etc. I'm liking it!

Reply:
Great to hear! Would you mind updating the PR title to mark this as a breaking change? I think this'll help anyone who uses `find_by_sql` in their observability backends know to update things accordingly. 

It can be marked as a breaking change by updating the title to start with `fix!:` instead of `fix:` 

If you don't think this should be a breaking change, let me know and we can discuss it further.

Reply:
Done! Totally agree that it's a breaking change.